samza-azure/src/main/java/org/apache/samza/AzureClient.java

```diff
@@ -47,7 +51,14 @@
   AzureClient(String storageConnectionString) {
     try {
       account = CloudStorageAccount.parse(storageConnectionString);
+
       blobClient = account.createCloudBlobClient();
+      // Set retry policy for operations on the blob. In this case, every failed operation on the blob will be retried thrice, after 5 second intervals.
```

Comment:
The exception handling is exactly the same for creating the client side references for Azure services. Also, this instance is only created once for the duration of the application and used accordingly. Hence, I believe it doesn't make sense to decouple the client creation. Let me know if you have more feedback on this.